server/routes/auth.js

```diff
@@ -0,0 +1,336 @@
+        if (password && (!authHelpers.comparePass(password, req.user.password))) {
+            const salt = bcrypt.genSaltSync();
+            const hash = bcrypt.hashSync(req.body.password, salt);
+            id = req.user.id
+            db.from('users')
+                .returning('id')
+                .where('id', '=', id)
+                .update({
+                    password: hash
+                })
+                .then((result) => {
+                    res.status(200).send({ 'status': 'ok', 'redirect': '/' });
+                    logger.auth.debug(req.username + 'Password Reset Successfully')
+                })
+                .catch((err) => {
+                    res.status(500).send({ 'status': 'failed', 'error': 'Failed to update password' });
+                    logger.auth.error(req.username + 'error resetting password' + err)
+                })
+        } else {
+            res.status(500).send({ 'status': 'failed', 'error': 'Password Blank or the Same' });
+        }
+    });
+
+router.route('/userCheck/username/:id')
+    .get(function (req, res, next) {
+        var id = req.params.id;
+        db.from('users')
+            .select('username')
+            .where('username', id)
+            .then((row) => {
+                if (row.length > 0) {
+                    row = row[0]
+                    res.status(200);
+                    res.json(row);
+                } else {
+                    row = {
+                        "username": "",
+                        "password": "",
+                        "givenname": "",
+                        "surname": "",
+                        "email": "",
+                        "role": "user",
+                        "status": "active"
+                    };
+                    res.status(200);
+                    res.json(row);
+                }
+            })
+            .catch((err) => {
+                logger.main.error(err);
+                return next(err);
+            })
+    });
+
+router.route('/userCheck/email/:id')
+    .get(function (req, res, next) {
+        var id = req.params.id;
+        db.from('users')
+            .select('email')
+            .where('email', id)
+            .then((row) => {
+                if (row.length > 0) {
+                    row = row[0]
+                    res.status(200);
+                    res.json(row);
+                } else {
+                    row = {
+                        "username": "",
+                        "password": "",
+                        "givenname": "",
+                        "surname": "",
+                        "email": "",
+                        "role": "user",
+                        "status": "active"
+                    };
+                    res.status(200);
+                    res.json(row);
+                }
+            })
+            .catch((err) => {
+                logger.main.error(err);
+                return next(err);
+            })
+    });
```

Comment:
This whole chunk is brute force risk, may need to reconsider... not sure best practice there as it's really annoying when sites only err out on submission, but equally bad if sites allow bad actors to run through common username lists to get brute force candidates

You're also returning the row there... so I could get /userCheck/username/admin and get the admin user's pw hash, their full name, their email address 😱

Reply:
No you can't, i'm only selecting the field i want...... (i think)

I had issues selecting the indivdual items after grabbing the whole row, so i made the select more specific

Reply:
I did think of the brute force while doing it, but i cant think of a better way to prevent conflicts other than bombing out with a sorry you fucked up when submitting them...

Reply:
Right yeah, `.select('email')`, the empty object threw me off lol

Reply:
I added some Brute-Force prevention with express-brute. 

Let me know what you think!

Reply:
Ooh that looks cool, you can probably be a bit more forgiving with the limits since it's something that would be a little spam heavy already. For exponential backoff the minimum can be something like 5 seconds just to filter out robots, and could do like 10 or 20 until it starts throttling